great_expectations/expectations/core/expect_column_values_to_be_in_set.py

```diff
@@ -411,15 +411,20 @@ def validate_configuration(
         self, configuration: Optional[ExpectationConfiguration]
     ) -> bool:
         super().validate_configuration(configuration)
+        # supports extensibility by allowing value_set to not be provided in config but captured via child-class default_kwarg_values, e.g. parameterized expectations
+        value_set = configuration.kwargs.get(
+            "value_set"
+        ) or self.default_kwarg_values.get(
+            "value_set"
+        )  # raises if default_kwarg_values is not overridden & value_set not provided
         try:
-            assert "value_set" in configuration.kwargs, "value_set is required"
-            assert (
-                isinstance(configuration.kwargs["value_set"], (list, set, dict))
-                or configuration.kwargs["value_set"] is None
-            ), "value_set must be a list, set, or None"
-            if isinstance(configuration.kwargs["value_set"], dict):
+            assert value_set != [] and value_set is not None, "value_set is required"
```

Comment:
Would it make sense to do `assert value_set` here instead? or `assert len(value_set) > 0`?

Reply:
Good question; I think the right pattern here is probably `assert "value_set" in configuration.kwargs or value_set`

This supports the new behavior without breaking our extant functionality of an empty `value_set` leading to a successful test. Now, the idea that column values *can be in* a non-existent set is not my favorite, but that feels like a bigger conversation 🙃